app/models/devise_token_auth/concerns/user.rb

```diff
@@ -125,7 +126,10 @@ def token_is_current?(token, client)
       DateTime.strptime(expiry.to_s, '%s') > Time.zone.now &&
 
       # ensure that the token is valid
-      DeviseTokenAuth::Concerns::User.tokens_match?(token_hash, token)
+      (
+        (token_hash.present? && DeviseTokenAuth::Concerns::User.tokens_match?(token_hash, token)) or
```

Comment:
I would move `DeviseTokenAuth::Concerns::User.tokens_match?` to a method here, so you don't repeat twice